frontend/src/main/web/lib/components/UserProfile.jsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import RecentContributions from './RecentContributions';
+import Configs from '../constants/Configs';
+import UserMatrixStore from '../stores/UserMatrixStore';
+
+var UserProfile = React.createClass({
+
+  getMatrixState: function() {
+    return UserMatrixStore.getMatrixState();
+  },
+
+  getInitialState: function() {
+    return this.getMatrixState();
+  },
+
+  componentDidMount: function() {
+    UserMatrixStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    UserMatrixStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    this.setState(this.getMatrixState());
+  },
+
+  render: function() {
+    var user = Configs.data.profileUser,
+      authenticated = Configs.user ? Configs.user.loggedIn : false,
+      recentContribution = (<div></div>),
+      imageUrl = '',
+      name = '',
+      languageTeams = '',
+      username;
+
+    if(authenticated) {
+      recentContribution = (
+          <div className="g__item w--3-4 w--1-2-m">
+            <div className="bg--pop-highest l--pad-v-1">
+              <RecentContributions
+                loading={this.state.loading}
+                dateRangeOption={this.state.dateRangeOption}
+                matrixForAllDays={this.state.matrixForAllDays}
+                wordCountsForSelectedDayFilteredByContentState={this.state.wordCountsForSelectedDayFilteredByContentState}
+                wordCountsForEachDayFilteredByContentState={this.state.wordCountsForEachDayFilteredByContentState}
+                contentStateOption={this.state.contentStateOption}
+                selectedDay={this.state.selectedDay}
+                dateRange={this.state.dateRange}/>
+            </div>
+          </div>
+      );
+    }
+
+    if(user !== null) {
+      username =  user.username;
+      imageUrl = user.imageUrl;
+      name = user.name;
+      languageTeams = user.languageTeams;
+    }
+
+    return (<div className="g">
```

Comment:
Maybe it just picked up something changed near it - I guess it would do that if any of the lines above that it shows are changed.